### ISSUE TITLE
Issue about accents and some characters

### ISSUE THREAD

**Issue by jbtatou** [WilldooIT/Pentaho-reports-for-OpenERP#183]
Hi, 

Thank you for the latest fixes for version 8.
I have a problem with the accents and such '°' from Odoo.
I do not have this problem when I get a prd bill.
Is it a mistake in the prd file or another.

thank you,
jb

**Comment by richard-willdooit** [comment]
@jbtatou Could you be more specific?  Do you have these characters in your source data?  Is the report an SQL or object based report?  Do the characters come through correctly when you preview the report running in the designer and only fail on deployment within Odoo?  Is the same font being picked up when used in the designer as when being used on Odoo deployment?  What if the report creates an xls file instead of a pdf?

**Comment by aitorbouzas** [comment]
@richard-willowit I've got the same issue, and it doesn't matter the output... If, for example, a product has an í (character with accent) and I try to print the report from a sale order with that product the line remains blank in HTML, PDF, plain text... In other cases, out from details block, the characters are just removed. For example: "Aquí" shows as "Aqu"

My report is object based, and these characters come through correctly in the preview of the report designer.

**Comment by angelmoya** [comment]
Same issue,

The label "Albarán" is shows as "Albarn", when print pdf in odoo, same problem with text-field.

In Pentaho Report Designer I fixed doing this:
1. File/Configuration/ output-pageable-pdf:  encoding must be set to UTF-8
2. Structure/ Master Report -> Style -> Font -> Family : Inherit false, family Arial

but same report print in odoo don't work.

**Comment by flotho** [comment]
Hy, 
Localisation is always a sensitive part of server conf. Didi you check that your dev environment has the same locale as the servers one?
Additionnally did you check that the values in the database have not been paste from a PDF. I frequently have issue with IMac and copy paste from PDF, invisible characters are displayed in a weird manner

**Comment by angelmoya** [comment]
Yes, Ubuntu 14.04 is installed in 3 systems (odoo server, java server, development system)
In all systems /etc/default/locale is:
LANG="es_ES.UTF-8"

Values in database are updated by odoo, and labels are written in PRD.

**Comment by richard-willdooit** [comment]
@AngelMoya-Domatix 

I am sorry, this is an area I am unable to assist with, as your answers seem to indicate everything is fine.  Is the report SQL based?  (My suspicion is it doesn't matter).

What happens if you put an accented character in the hard coded text part of the report?  If it generates correctly, then it is more likely to be a data retrieval / handling issue.  If not, then it is more likely to be a fonts and settings issue.

**Comment by aitorbouzas** [comment]
As I said in #205 I just found out that in HTML the report is printing now well... The problem could be in the process that generates the PDF?

**Comment by aitorbouzas** [comment]
Now I found out that it's a font issue... The pdf output does not gives the same font as the HTML one (the HTML one gives the same that in the designer)

EDIT:
Also, i've installed the font i'm using in the designer into my Ubuntu 14.04 Server (Calibri) and tried some others without success.

**Comment by aitorbouzas** [comment]
And also, the fonts are in /usr/share/fonts

**Comment by aitorbouzas** [comment]
Did it! After an installation of new fonts and a reboot everything worked well.

Just did this for installing liberation fonts:
sudo apt-get install ttf-liberation

Then reboot and use Liberation Sans in your designer (if it's Windows, you must install the fonts before designing the report)

**Comment by RickHunterEc** [comment]
Hi @AngelMoya-Domatix. I solved this problem with this link:

http://forums.pentaho.com/showthread.php?77127-Problema-con-tildes-en-Consola-usuario-3-6-1

Steps:
1) Change encoding in File / Configuration / select section "output-pageable-pdf", "Encoding" and change to "ISO-8859-1".
2) When design in PRD, you can this option permanent, add this line in file  "resources/classic-engine.properties" of PRD:

`org.pentaho.reporting.engine.classic.core.modules.output.pageable.pdf.Encoding=ISO-8859-1`

You can select any fonts when you export report in pentaho.
